### DISCUSSION TITLE
Lesson 6: Compiling ERROR (6359) - Deploying to a local chain III

### DISCUSSION THREAD

**Discussion by 0xTagger** [Cyfrin/foundry-full-course-cu#500] [Q&A]
Hi, I am experiencing a compiling error during Lesson 6 - Deploying to a local chain III. The error is as follows: **_Error (6359): Return argument type type(contract SimpleStorage) is not implicitly convertible to expected type (type of first return variable) contract SimpleStorage._**

I have re-written the code several times as well as copied the exact code from the repo, but the issue remains the same. Please see the pictures below for full details.

<img width="912" alt="Screenshot 2023-07-31 at 11 03 54 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/6f64e228-b5da-430b-b727-791a2bec9a7c">
<img width="1325" alt="Screenshot 2023-07-31 at 11 06 01 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/c446e6a2-5555-4c30-825a-7aced64ae63f">

I have noticed when using import that both files "import {Script} from" AND "import {SimpleStorage} from" do not change color and remain white. The same goes for SimpleStorage throughout the DeploySimpleStorage.s.sol contract.

Any feedback is much appreciated.

**Comment by shikhar229169** [comment]
I also faced this issue sometimes and restarting vs code fixed it for me.

**Comment by Niferu** [comment]
Hello @0xTagger 

Try to click and open `foundry.toml` then go back to your `DeploySimpleStorage.s.sol` and if nothing changed just click in code at last spot of `pragma solidity 0.8.19;` and change it to `pragma solidity 0.8.18;` for a second save and change it back, it should fix the issue, if not just use VSCode reload window or restart VSCode.

**Reply by 0xTagger** [reply]
Thank you @shikhar229169 and @Neftyr for the input! After trying your suggestions I restarted VSCode and a pop-up displayed on my screen showing that I was running two Solidity extensions. This was the cause for error and once I disabled one of the extensions everything compiled successfully.

**Comment by 0xDevGod** [comment]
Hey @0xTagger 👋 

Add the complete path like this 
```
import {Script} from "lib/forge-std/src/Script.sol";

```

**Comment by theirrationalone** [comment]
Try to restart your vs-code... 

If after restarting vs-code still You're facing same error then Please make sure You're Using Right extension for solidity support. 

There's clearly mentioned in the error: import callback not supported. 

It happens when we try to have a bad Or more than one extension in use. 

Make sure to use right extension for solidity formatting support. Or you can use `forge fmt` only for formatting solidity code and best extension that i use to have support is `Nomicfoundation`'s extension also suggested by Patrick (SIR).
